src/coprocessor/codec/mysql/duration.rs

```diff
@@ -464,6 +647,27 @@ mod tests {
             (b"00:00:00.777777", 2, Some("00:00:00.78")),
             (b"00:00:00.777777", 6, Some("00:00:00.777777")),
             (b"00:00:00.001", 3, Some("00:00:00.001")),
+            // NOTE: The following case is easy to fail.
+            (b"- 1 ", 0, Some("-00:00:01")),
+            (b"1:2:3", 0, Some("01:02:03")),
+            (b"1 1:2:3", 0, Some("25:02:03")),
+            (b"-1 1:2:3.123", 3, Some("-25:02:03.123")),
+            (b"-.123", 3, Some("-00:00:00.123")),
+            (b"12345", 0, Some("01:23:45")),
+            (b"-123", 0, Some("-00:01:23")),
+            (b"-23", 0, Some("-00:00:23")),
+            (b"- 1 1", 0, Some("-25:00:00")),
+            (b"-1 1", 0, Some("-25:00:00")),
```

Comment:
The above two cases are incompatible with MySQL 8.0, maybe we should use an issue to track it.

Reply:
OK! 😃